engines/native/system/process_darwin.go

```diff
@@ -0,0 +1,204 @@
+	// Close all connected streams, ignore errors
+	if p.stdin != nil {
+		p.stdin.Close()
+	}
+	if p.stdout != nil {
+		p.stdout.Close()
+	}
+	if p.stderr != nil && p.stderr != p.stdout {
+		p.stderr.Close()
+	}
+	if p.pty != nil {
+		p.pty.Close()
+	}
+
+	// Resolve with result
+	p.resolve.Do(func() {
+		p.result = err == nil
+	})
+}
+
+// Wait for process to terminate, returns true, if exited zero.
+func (p *Process) Wait() bool {
+	p.resolve.Wait()
+	return p.result
+}
+
+// Kill the process
+func (p *Process) Kill() {
+	p.cmd.Process.Kill()
+}
+
+// SetSize of the TTY, if running as TTY or do nothing.
+func (p *Process) SetSize(columns, rows uint16) {
+	if p.pty != nil {
+		p.pty.SetSize(columns, rows)
+	}
+}
+
+// StartProcess starts a new process with given arguments, environment variables,
+// and current working folder, running as given user.
+//
+// Returns an human readable error explaining why the sub-process couldn't start
+// if not successful.
+func StartProcess(options ProcessOptions) (*Process, error) {
+	// Default arguments to system shell
+	if len(options.Arguments) == 0 {
+		options.Arguments = []string{defaultShell}
+	}
+
+	// If WorkingFolder isn't set find home folder of options.Owner (if set)
+	// or current user
+	if options.WorkingFolder == "" {
+		if options.Owner != nil {
+			options.WorkingFolder = options.Owner.homeFolder
+		} else {
+			u, err := user.Current()
+			if err != nil {
+				panic(fmt.Sprintf("Failed to lookup current user, error: %s", err))
+			}
+			options.WorkingFolder = u.HomeDir
+		}
+	}
+
+	// Default stdout to os.DevNul
+	if options.Stdout == nil {
+		options.Stdout = ioext.WriteNopCloser(ioutil.Discard)
+	}
+
+	// Default stderr to stdout
+	if options.Stderr == nil {
+		options.Stderr = options.Stdout
+	}
+
+	// Create process and command
+	p := &Process{}
+	p.cmd = exec.Command(options.Arguments[0], options.Arguments[1:]...)
+	p.cmd.Env = formatEnv(options.Environment)
+	p.cmd.Dir = options.WorkingFolder
+
+	// Set owner for the process
+	if options.Owner != nil {
+		p.cmd.SysProcAttr = &syscall.SysProcAttr{
+			Credential: &syscall.Credential{
+				Uid: options.Owner.uid,
+				Gid: options.Owner.gid,
+			},
+		}
+	}
+
+	// Start the process
+	var err error
+	if !options.TTY {
+		p.cmd.Stdin = options.Stdin
+		p.cmd.Stdout = options.Stdout
+		p.cmd.Stderr = options.Stderr
+		p.stdin = options.Stdin
+		p.stdout = options.Stdout
+		p.stderr = options.Stderr
+		err = p.cmd.Start()
+	} else {
+		p.pty, err = pty.Start(p.cmd)
+		if options.Stdin != nil {
+			p.sockets.Add(1)
+			go func() {
+				io.Copy(p.pty, options.Stdin)
+				p.sockets.Done()
+				// Kill process when stdin ends (if running as TTY)
+				p.Kill()
+			}()
+		}
+		p.sockets.Add(1)
+		go func() {
+			ioext.CopyAndClose(options.Stdout, p.pty)
+			p.sockets.Done()
+		}()
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("Unable to execute binary, error: %s", err)
+	}
+
+	// Go wait for result
+	go p.waitForResult()
+
+	return p, nil
+}
```

Comment:
Is this a modified version of the system package library? How come we can't use the standard library natively? (I believe it has support for running processes as a different user on darwin/linux - even if not on Windows)